src/model/guild/mod.rs

```diff
@@ -725,11 +725,13 @@ impl Guild {
     ///
     /// [`Member`]: struct.Member.html
     pub fn member_named(&self, name: &str) -> Option<&Member> {
-        let (name, discrim) = if let Some(pos) = name.find('#') {
-            let split = name.split_at(pos);
+        let (name, discrim) = if let Some(pos) = name.rfind('#') {
+            let split = name.split_at(pos+1);
+            
+            let split2 = (split.0.get(0..split.0.len()-1).unwrap() , split.1);
```

Comment:
Could we take advantage of this Option and instead of unwrapping, map over it instead? Even though it's in theory safe to unwrap, it'd still be nice not to.